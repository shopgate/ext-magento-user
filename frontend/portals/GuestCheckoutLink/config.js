/**
 * Config mapper
 */
/* eslint import/no-unresolved: 0 */
import config from '../../config';

export default config;
