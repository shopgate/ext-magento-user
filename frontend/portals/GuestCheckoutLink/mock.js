/**
 * Mocked state with redirectLocation to checkout page.
 * @type {Object}
 */
export const mockedStateFromCartView = {
  history: {
    pathname: '/',
    redirectLocation: {
      pathname: '/checkout',
    },
  },
};

/**
 * Mocked state with redirectLocation to more page.
 * @type {Object}
 */
export const mockedStateFromMyAccountView = {
  history: {
    pathname: '/',
    redirectLocation: {
      pathname: '/more',
    },
  },
};

/**
 * Mocked state with redirectLocation to home page.
 * @type {Object}
 */
export const mockedStateFromAnyOtherView = {
  history: {
    pathname: '/',
    redirectLocation: {
      pathname: '/',
    },
  },
};
