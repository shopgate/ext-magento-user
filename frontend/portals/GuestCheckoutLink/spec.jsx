import React from 'react';
import { mount } from 'enzyme';
import {
  mockedContextFromCartView,
  mockedContextFromMyAccountView,
  mockedContextFromAnyOtherView,
} from './mock';

import GuestCheckoutLink from './index';

jest.mock('./config', () => ({
  getUserAccountSettings: 1,
}));

jest.mock('@shopgate/pwa-common/components/Link');

beforeEach(() => {
  jest.resetModules();
});

/**
 * Creates component with provided context state.
 * @param {Object} mockedContext Mocked context.
 * @return {ReactWrapper}
 */
const createComponent = (mockedContext) => {
  const mockContext = jest.fn();
  mockContext.mockReturnValue(mockedContext);
  jest.mock('@shopgate/pwa-common/context', () => ({
    RouteContext: {
      Consumer: ({ children }) => children(mockContext()),
    },
  }));

  return mount(<GuestCheckoutLink />);
};

describe('<GuestCheckoutLink />', () => {
  it('should render with one link for login in checkout page', () => {
    const component = createComponent(mockedContextFromCartView);
    expect(component).toMatchSnapshot();
    expect(component.find('Connect(Link)').exists()).toBe(true);
  });

  it('should render null in case of default login', () => {
    const component = createComponent(mockedContextFromMyAccountView);
    expect(component).toMatchSnapshot();
    expect(component.find('Connect(Link)').exists()).toBe(false);
  });

  it('should render null in case of all other', () => {
    const component = createComponent(mockedContextFromAnyOtherView);
    expect(component).toMatchSnapshot();
    expect(component.find('Connect(Link)').exists()).toBe(false);
  });
});
