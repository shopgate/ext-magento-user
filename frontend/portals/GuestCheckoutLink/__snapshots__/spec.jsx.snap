// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestCheckoutLink /> should render null in case of all other 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(GuestCheckoutLink)>
    <GuestCheckoutLink
      dispatch={[Function]}
      redirect="/"
    />
  </Connect(GuestCheckoutLink)>
</Provider>
`;

exports[`<GuestCheckoutLink /> should render null in case of default login 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(GuestCheckoutLink)>
    <GuestCheckoutLink
      dispatch={[Function]}
      redirect="/more"
    />
  </Connect(GuestCheckoutLink)>
</Provider>
`;

exports[`<GuestCheckoutLink /> should render with one link for login in checkout page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(GuestCheckoutLink)>
    <GuestCheckoutLink
      dispatch={[Function]}
      redirect="/checkout"
    >
      <div
        className="css-nil"
      >
        <Translate
          className="css-1fzm7bm"
          params={Object {}}
          string="checkout.or"
        >
          <span
            className="css-1fzm7bm"
          >
            checkout.or
          </span>
        </Translate>
        <Link
          className="css-13v1itq"
          disabled={false}
          href="/checkout_guest"
        >
          <div
            aria-hidden={true}
            className="css-c6ki83 css-13v1itq"
            data-test-id="link: /checkout_guest"
            onClick={[Function]}
            role="link"
          >
            <Translate
              className=""
              params={Object {}}
              string="checkout.continue_as_guest"
            >
              <span
                className=""
              >
                checkout.continue_as_guest
              </span>
            </Translate>
            .
          </div>
        </Link>
      </div>
    </GuestCheckoutLink>
  </Connect(GuestCheckoutLink)>
</Provider>
`;
